api/src/api.ts

```diff
@@ -16,36 +37,48 @@ export const plugin = {
   name: "API",
   version: "0.0.2",
   register: async function (server: Hapi.Server, options: ApiOptions) {
-    const { connection: dbConnection } = options;
+    const { connection: dbConnection, config } = options;
 
     server.route({
       method: "GET",
       path: "papers/list",
       handler: async (request) => {
-        let offset;
-        if ('offset' in request.query) {
-          const o = parseInt(
-            Array.isArray(request.query.offset)
-              ? request.query.offset[0]
-              : request.query.offset
-          );
-          if (!isNaN(o)) {
-            offset = o;
+        const offset = firstIntOrDefault(request, "offset", 0);
+        const size = firstIntOrDefault(request, "size", 25);
+
+        const papers = await dbConnection.getAllPapers(offset, size);
+        const paperIds = papers.rows.map(p => p.s2_id);
+
+        // We fetch metadata about each paper (it's title, author names, etc) from S2's public
+        // API and merged them into the result set.
+        const s2PaperInfoByPaperId: { [pid: string]: Paper } = {};
+        for (const s2Paper of await s2Api.getPapers(paperIds, config.s2.apiKey, true)) {
```

Comment:
same here

Reply:
Yes and no.

In this case I don't think we need to, because I meant to set `failOnError` to `false`.  This causes the function we're calling to not fail *everything* if a single request fails.

I'll fix that, and add a `try / catch` above. Thanks!